raiden-network/light-client
# Channel balance-based Monitoring threshold

Fixes #1369 

**Short description**
Implements a new config value for the SDK: `rateToSvt`, which holds a `token => wei` mapping. Only tokens present on this mapping will have monitoring enabled, and monitoring will be requested (i.e. paid to MS as rewards) only after channel's received total (off and on-chain unlocked sum), normalized through this rate, surpasses `monitoringReward` config, which default now is set to `5 SVT` (same as raiden-py,  hopefully enough for a MS to pick up our request, may be tweaked if needed). The `rateToSvt` config is empty by default, meaning receiving is still disabled by default and therefore no monitoring is applied.

**Definition of Done**


**Steps to manually test the change (dApp)**

1. In served dApp, set the rate:
```ts
// rate=5e18 means 1e18wei of TKN (1TKN if decimals=18) is worth 5SVT, so 1+TKN is needed to trigger monitoring
raiden.updateConfig({ rateToSvt: { "<token_addr>": "5000000000000000000" } });
```
2. Receiving gets enabled
3. In another dApp account, transfer 0.9TKN to first client. Transfer should be received, but no monitoring signatures are requested
4. Make new transfer, worth 0.11TKN; threshold cross value of default monitoringReward of 5SVT, monitoring prompt happens 15s later and is sent to global rooms: `messageGlobalSend` actions to `meta: { roomName: "raiden_goerli_monitoring" }` containing the `type: "RequestMonitoring"` messages.
